improver/calibration/reliability_calibration.py

```diff
@@ -1441,3 +1443,60 @@ def process(
         calibrated_forecast.data = calibrated_forecast.data.astype("float32")
 
         return calibrated_forecast
+
+
+class EnforceConsistentProbabilities(PostProcessingPlugin):
+    """Reduces the probabilities of a forecast to make it consistent
+    with another provided forecast.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialise class for enforcing probabilities between two forecasts.
+        """
+        self.diff_for_warning = 0.3
+
+    def process(self, forecast_cube: Cube, ref_forecast: Cube) -> Cube:
+        """
+        Lowers the probabilities of the forecast_cube to make it consistent with
+        the reference forecast. This is done by lowering the probabilities of the forecast cube,
+        to be equal to the reference forecast if they are higher than the reference forecast.
+        If the probability is already less than or equal to the reference forecast then the
+        probability is not altered.
+
+        Args:
+            forecast_cube:
+                A forecast cube of probabilities
+            ref_forecast:
+                A cube of probabilities that is used as an upper limit for
+                the forecast_cube probabilities. It must have the same dimensions
+                as the forecast_cube.
+
+        Returns:
+            A forecast cube of probabilities that has identical metadata to forecast_cube but
+            with reduced probabilities to make it consistent with ref_forecast
+        """
+
+        assert_spatial_coords_match([forecast_cube, ref_forecast])
+
+        # puts ref_forecast's data into a cube with the same metadata as forecast_cube so they
+        # can be combined
+        ref_forecast_metadata = forecast_cube.copy(data=ref_forecast.data)
+
+        diff = Combine(operation="-")([ref_forecast_metadata, forecast_cube])
```

Comment:
I've removed assert spatial coordinates match.

It's not quite true that Combine ignores all metadata. It compares the coordinate points and metadata so if the coordinates have different names, even if they are the same shape it raises an error.  This is a problem with the clouds because it trys to compare "cloud_area_fraction" to "low_and_medium_type_cloud_area_fraction" and can't do it so errors get raised. I have managed to avoid updating all the metadata by renaming the threshold coordinate which seems to do the job.